### PR TITLE
[ci] Remove default value & use environment variables for release automation pipeline

### PR DESCRIPTION
**Pull request by khluu**
## Changes
- Eliminate use of default value for `RAY_VERSION` and `RAY_HASH` to ensure that correct version is manually put in and sanity checked. 
- Remove default value for Python version.
- Export env variables to linux sanity check script.

## Checks
- Build passed on Buildkite with ray version & hash set to `2.9.2` : https://buildkite.com/ray-project/release-automation/builds/116

### CODE REVIEW

File: .buildkite/release-automation/verify-macos-wheels.sh [hunk 1]
```diff
@@ -15,18 +15,18 @@ if [[ $# -ne 1 ]]; then
...
-export RAY_VERSION="${RAY_VERSION:-2.9.1}"
-export RAY_HASH="${RAY_HASH:-cfbf98c315cfb2710c56039a3c96477d196de049}"
+export RAY_VERSION="${RAY_VERSION}"
+export RAY_HASH="${RAY_HASH}"
 export USE_BAZEL_VERSION="${USE_BAZEL_VERSION:-5.4.1}"
 
 install_bazel() {
     if [[ "${mac_architecture}" = "arm64" ]]; then
-      URL="https://github.com/bazelbuild/bazelisk/releases/download/${BAZELISK_VERSION}/bazelisk-darwin-arm64"
+        URL="https://github.com/bazelbuild/bazelisk/releases/download/${BAZELISK_VERSION}/bazelisk-darwin-arm64"
```
Comment (can-anyscale) [inline .buildkite/release-automation/verify-macos-wheels.sh:24]: nit: remove indentation?

File: .buildkite/release-automation/wheels.rayci.yml [hunk 2]
```diff
@@ -8,6 +8,8 @@ steps:
     job_env: forge
     commands:
       - export PYTHON_VERSION={{matrix}}
+      - export RAY_VERSION="$RAY_VERSION"
```
Comment (can-anyscale) [inline .buildkite/release-automation/wheels.rayci.yml:11]: is RAY_VERSION set part of the build environment? if so we don't need export it again

File: release/util/sanity_check_cpp.sh [hunk 3]
```diff
@@ -5,3 +5,4 @@ set -e
 rm -rf ray-template
 ray cpp --generate-bazel-project-template-to ray-template
 pushd ray-template && bash run.sh
+echo "Sanity check succeeded for C++."
```
Comment (can-anyscale) [inline release/util/sanity_check_cpp.sh:8]: put in another PR?

File: .buildkite/release-automation/verify-linux-wheels.sh [hunk 4]
```diff
@@ -4,9 +4,9 @@ set -euo pipefail
 
 set -x
 
-export PYTHON_VERSION="${PYTHON_VERSION:-3.8}"
-export RAY_VERSION="${RAY_VERSION:-2.9.1}"
-export RAY_HASH="${RAY_HASH:-cfbf98c315cfb2710c56039a3c96477d196de049}"
+export PYTHON_VERSION="${PYTHON_VERSION}"
+export RAY_VERSION="${RAY_VERSION}"
+export RAY_HASH="${RAY_HASH}"
```
Comment (can-anyscale) [inline .buildkite/release-automation/verify-linux-wheels.sh:9]: maybe throw/fail if these RAY_VERSION/RAY_HASH value is empty

**Review by can-anyscale — CHANGES_REQUESTED** [review]
My main question is to safeguard the pipeline in case a human makes mistakes and supply invalid/empty value for ray version/hash

File: .buildkite/release-automation/wheels.rayci.yml (hunk 2 above)
Comment (khluu) [reply .buildkite/release-automation/wheels.rayci.yml]: `RAY_VERSION` and `RAY_HASH` should be put in as environment variables when users create new build on Buildkite. This build, for example, is failing when `RAY_VERSION` is not exported in the step commands: fce7a361807580953364e2da964f9498f3123bf9

File: .buildkite/release-automation/verify-linux-wheels.sh (hunk 4 above)
Comment (khluu) [reply .buildkite/release-automation/verify-linux-wheels.sh:18]: done!

File: release/util/sanity_check_cpp.sh (hunk 3 above)
Comment (khluu) [reply release/util/sanity_check_cpp.sh]: done!

File: .buildkite/release-automation/verify-linux-wheels.sh (hunk 4 above)
Comment (can-anyscale) [reply .buildkite/release-automation/verify-linux-wheels.sh:18]: If RAY_VERSION, etc. are already part of the global environment, do we need to export it again?

**Comment by khluu** [comment]
@can-anyscale  Reverted it back to use `bash -i` and export build env vars into the linux script now. https://buildkite.com/ray-project/release-automation/builds/126 passed for all envs.

**Review by can-anyscale — APPROVED** [review]

File: .buildkite/release-automation/verify-linux-wheels.sh (hunk 4 above)
Comment (aslonnie) [reply .buildkite/release-automation/verify-linux-wheels.sh]: we do, otherwise it won't be visible in sub commands.
